lib/spack/spack/database.py

```diff
@@ -77,6 +77,13 @@ def converter(self, spec_like, *args, **kwargs):
     return converter
 
 
+_valid_install_states = [
```

Comment:
I think promoting this to an enum like

```
class InstallState(object):
  INSTALLED = 'installed'
  DEPRECATED = 'deprecated'
  ...
```

Can avoid typos and also help users find where this is defined and how it is handled if referenced accordingly in other modules (vs. a "disconnected" reference to a static string like "installed").

Reply:
I'm trying this out. Hopefully by tomorrow I'll have an implementation up, it looks like it's close to working.